File: code/graphs/bfs.cpp
void bfs(Graph& g, int s) {
    queue<int> q; q.push(s);
    vector<bool> seen(g.V, false); seen[s] = true;
    while (!q.empty()) {
        int u = q.front(); q.pop();     // visit u
        for (int v : g.adj[u])
            if (!seen[v]) {
                seen[v] = true;         // mark on ENQUEUE
                q.push(v);
            }
    }
}

File: code/graphs/bfs.java
void bfs(Graph g, int s) {
    Queue<Integer> q = new ArrayDeque<>();
    q.add(s);
    boolean[] seen = new boolean[g.V];
    seen[s] = true;
    while (!q.isEmpty()) {
        int u = q.poll();               // visit u
        for (int v : g.adj.get(u))
            if (!seen[v]) {
                seen[v] = true;         // mark on ENQUEUE
                q.add(v);
            }
    }
}
